granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -88,13 +198,46 @@ def get_cgroup_mount(controller: str, resolve_host_root_links: bool = True) -> O
     v2_path = find_v2_hierarchy(resolve_host_root_links)
     if v1_paths and controller in v1_paths:
         # Either v1 or hybrid with the requested controller bound to a v1 hierarchy
-        return v1_paths[controller]
+        return CgroupCoreV1(Path(v1_paths[controller]))
     if v2_path:
-        # v2 (unified) - all controllers are in a single unified hierarchy
-        return v2_path
+        # v2 (unified) - check given controller is supported in cgroup v2
+        cgroup_v2 = CgroupCoreV2(Path(v2_path))
+        if cgroup_v2.is_controller_supported(controller):
+            return cgroup_v2
     # No cgroup mount of the requested controller
     return None
 
 
-def is_known_controller(controller: str) -> bool:
-    return controller in SUBSYSTEMS
+def get_cgroup_mount_checked(controller: str, resolve_host_root_links: bool = True) -> CgroupCore:
+    cgroup_mount = get_cgroup_mount(controller)
+    assert cgroup_mount is not None, f"Could not find cgroup mount point for controller {controller!r}"
+    return cgroup_mount
+
+
+def create_cgroup_from_path(controller: str, cgroup_path: Path) -> CgroupCore:
+    cgroup_mount = get_cgroup_mount_checked(controller)
```

Comment:
Glossary wise - let's create consistent naming here. `cgroup_path` here might be relative to the controller and might already include the controller & mount, right?

Let's create standard names:
* `cgroup_full_path` - mount+controller+path
* `cgroup_path` - path on top of the controller mount
* `controller` - just the controller name
* `cgroup_mount` - just the mount
Makes sense?

Then this function becomes:
```
def create_cgroup_from_path(controller: str, cgroup_path_or_full_path: Path):
    try:
        cgroup_path_or_full_path.relative_to(cgroup_mount.path)
        # it's a full path
        cgroup_full_path = cgroup_path_or_full_path
    except ValueError:
        # it's path relative to the controller
        cgroup_full_path = cgroup_mount.path / cgroup_path
```

And fix other functions accordingly. Makes it easier to browse through the code if you use consistent names, since things don't have types and a `Path` can be either.

Alternatively, you can create types for them... but I think names suffice.